pkg/steps/template.go

```diff
@@ -382,21 +382,28 @@ func isPodCompleted(podClient coreclientset.PodInterface, name string) (bool, er
 }
 
 func waitForTemplateInstanceReady(templateClient templateclientset.TemplateInstanceInterface, instance *templateapi.TemplateInstance) (*templateapi.TemplateInstance, error) {
-	for {
-		ready, err := templateInstanceReady(instance)
-		if err != nil {
-			return nil, fmt.Errorf("could not determine if template instance was ready: %v", err)
+	var actualErr error
+	err := wait.PollImmediate(2*time.Second, 1*time.Minute, func() (bool, error) {
+		ready, actualErr := templateInstanceReady(instance)
+		if actualErr != nil {
+			log.Printf("could not determine if template instance was ready: %v", actualErr)
+			return false, nil
 		}
 		if ready {
-			return instance, nil
+			return true, nil
 		}
 
-		time.Sleep(2 * time.Second)
-		instance, err = templateClient.Get(instance.Name, meta.GetOptions{})
-		if err != nil {
-			return nil, fmt.Errorf("unable to retrieve existing template instance: %v", err)
+		instance, actualErr = templateClient.Get(instance.Name, meta.GetOptions{})
+		if actualErr != nil {
+			log.Printf("unable to retrieve existing template instance: %v", actualErr)
+			return false, nil
 		}
+		return true, nil
+	})
+	if err == nil {
```

Comment:
nit: you could drop this check and just 


```go
return instance, actualErr
``` 

The caller has to check `err == nil` on their end anyway